### ISSUE TITLE
Large number of command buffer aborts

### ISSUE THREAD

**Issue by cmccuewg** [KhronosGroup/MoltenVK#831]
I'm having a lot of problems with GPU errors from my game application and unit test apps which use Vukan via MoltenVK.  These apps pass validation, but at runtime issue a handful of "GPU timeout" errors (IOAF code 2) followed by a ton of "ignored for causing prior/excessive GPU errors" (IOAF code 4).

These errors happen fairly reliably (maybe 80% of the time) in my unit test application, but the exact timing of them varies from run to run.

I found [this issue](https://github.com/KhronosGroup/MoltenVK/issues/602) which had similar problems, but it's been marked as closed and I'm already on the version of MoltenVK from the latest Vulkan SDK which appears to be 1.0.39.  That issue also had quite a bit of back-and-forth with different options, so it's not clear to me what from that thread is still valid and what I should try.

Could I get some guidance on how to figure out what might be going wrong here?  I can report back any results that I discover.

Thanks!

**Comment by cmccuewg** [comment]
I should also note that these applications run correctly on other non-MVK platforms.

**Comment by cdavis5e** [comment]
You're not running with `MVK_ALLOW_METAL_EVENTS=true`, are you? If not, then I'm afraid we'll need more information. Can you provide us with something that reproduces the problem?

**Comment by cmccuewg** [comment]
I don't believe that I am running with that option.  I'm on the Mac (though i believe this happens on iOS also), and I'm not doing anything other than including "vk_mvk_moltenvk.h" and linking to the static libMoltenVK.a from VulkanSDK 1.1.130.  

I can't post the source because even our unit tests require too much of our game engine.  Is there a version of vktrace that runs on the Mac?  It doesn't look like the SDK includes one.  Let me know what other options you might have in mind.
